Generate Visual Recognition V4

mediumTaj opened the pull request:
### Summary
This pull request adds Visual Recognition v4 support to the .NET SDK

lpatino10 commented on src/IBM.Watson.VisualRecognition.v4/VisualRecognitionService.cs:
```diff
@@ -0,0 +1,996 @@
+        /// You can also include images with the **images_file** parameter. (optional)</param>
+        /// <param name="threshold">The minimum score a feature must have to be returned. (optional)</param>
+        /// <returns><see cref="AnalyzeResponse" />AnalyzeResponse</returns>
+        public DetailedResponse<AnalyzeResponse> Analyze(string collectionIds, string features, List<FileWithMetadata> imagesFile = null, List<string> imageUrl = null, float? threshold = null)
```
`collectionIds` and `features` should both be lists.

lpatino10 commented on test/VisualRecognition.v4.IntegrationTests/Properties/AssemblyInfo.cs:
```diff
@@ -0,0 +1,19 @@
+// associated with an assembly.
+[assembly: AssemblyConfiguration("")]
+[assembly: AssemblyCompany("IBM")]
+[assembly: AssemblyProduct("IBM.Watson.VisualRecognition.v3.IntegrationTests")]
```
This one's for v4

lpatino10 commented on test/VisualRecognition.v4.UnitTests/VisualRecognitionUnitTests.cs:
```diff
@@ -0,0 +1,1491 @@
+        #endregion
+
+        #region Environments
+        //#region List Environments
```
Looks like everything under this has to do with Discovery haha

mediumTaj replied:
ugh... my unit tests got lost somewhere :(

lpatino10 replied:
Nooo 😭😭

mediumTaj replied:
:( seriously... thats what i get for working on multiple machines. Ill try to find them. If not, ill include them in unit test generation!

lpatino10 commented on examples/IBM.Watson.Assistant.v2.Examples/IBM.Watson.Assistant.v2.Examples.csproj:
```diff
@@ -6,11 +6,10 @@
 
   <ItemGroup>
-    <PackageReference Include="IBM.Cloud.SDK.Core" Version="1.0.0-rc9" />
+    <PackageReference Include="IBM.Cloud.SDK.Core" Version="1.0.0-rc91" />
```
This version number seems like it really jumped 😂

mediumTaj replied:
lol yeah nuget didn't like `rc10`... in hindsight i should have started with `rc01`

mediumTaj commented:
@lpatino10 I think I addressed all of the comments. Regarding unit tests, it looks like I might have misplaced them - possibly on my machine at home. I updated the tests for now and will look for them tonight. If they are not there I think update these as part of our unit test generation activities.

lpatino10 commented:
Changes look good to me but there are some CI errors 😕  I'll re-check once those are passing

mediumTaj commented:
:tada: This PR is included in version 4.0.0 :tada:

The release is available on [GitHub release](https://github.com/watson-developer-cloud/dotnet-standard-sdk/releases/tag/v4.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
